src/Mod/PartDesign/App/FeatureLoft.cpp

```diff
@@ -73,21 +74,32 @@ App::DocumentObjectExecReturn *Loft::execute()
 {
     auto getSectionShape =
         [](App::DocumentObject* feature, const std::vector<std::string> &subs) -> TopoDS_Shape {
-            if (!feature ||
-                !feature->isDerivedFrom(Part::Feature::getClassTypeId()))
+            if (!feature)
                 throw Base::TypeError("Loft: Invalid profile/section");
 
+            App::DocumentObject* linkedObj = feature;
+            // If the object is a link try to resolve it first.
+            if (feature->isDerivedFrom<App::Link>()) {
+                linkedObj = static_cast<App::Link*>(feature)->getLinkedObject(true);
+                if (!linkedObj)
+                    throw Base::TypeError("Loft: Invalid profile/section");
+            }
+
+            if (!linkedObj->isDerivedFrom<Part::Feature>())
+                throw Base::TypeError("Loft: All sections need to be part features");

```

Comment:
It looks like it's possible this will get displayed to the user in a dialog. Can you please wrap it in translation code?